src/CodeGen_GPU_Dev.cpp

```diff
@@ -149,38 +149,76 @@ void CodeGen_GPU_C::visit(const Shuffle *op) {
             internal_assert(op->vectors[0].type() == op->vectors[i].type());
         }
         internal_assert(op->type.lanes() == (int)op->indices.size());
-        const int max_index = (int)(op->vectors[0].type().lanes() * op->vectors.size());
+        std::vector<int> vector_first_index;
+        int max_index = 0;
+        for (const Expr &v : op->vectors) {
+            vector_first_index.push_back(max_index);
+            max_index += v.type().lanes();
+        }
         for (int i : op->indices) {
             internal_assert(i >= 0 && i < max_index);
         }
 
         std::vector<std::string> vecs;
         for (const Expr &v : op->vectors) {
-            vecs.push_back(print_expr(v));
+            std::string expr = print_expr(v);
+            internal_assert(!expr.empty()) << "Expression did not serialize.";
```

Comment:
nit: it's probably useful to print the offending Expr, `v` in the assert message.
```
internal_assert(!expr.empty()) << "Expression did not serialize: " << v;
```

Reply:
This assert should probably go. Expressions always serialize. I was seeing some odd stuff while working on it, and thought that I had an empty expression, which is just impossible. I'll remove this assert.

Reply:
Deleted it.